### DISCUSSION TITLE
Is it possible to change path of QML files?

### DISCUSSION THREAD

**Discussion by dphaldes** [KDAB/cxx-qt#884] [Q&A]
When loading qml files using the QMLModule Struct in CxxQtBuilder, the path of each qml file becomes 
`qrc:/qt/qml/{uri/of/module}/{path/to/file}` which is extremely long is some cases. I am learning QT right now, and it seems QT5 used a specific qrc file and would allow alias for files. Is this still possible with the new QML module stuff in QT 6 ?

**Comment by ahayzen-kdab** [comment]
For QML files normally you include the uri and then use the component directly.

Eg a file like `qrc:/qt/qml/com.kdab.ui/MyButton.qml` would be used like
```
import com.kdab.ui 1.0
MyButton {}
```

You may have a few places in C++ where you need to refer to the path, eg when setting the `main.qml` for `QQuickView`, but this tends to be minimal. So unsure if the `qml_files` needs support for this.

However for qrc files, like images, these could be used in multiple places and tend to have aliases to make things shorter.

So we should explore whether adding alias support in the `QmlModule` `qrc_files` is possible.

**Comment by dphaldes** [comment]
> qrc:/qt/qml/com.kdab.ui/MyButton.qml

Are you sure module name/ URI uses dots and not slashes ? Docs definitely say otherwise https://docs.rs/cxx-qt-build/latest/cxx_qt_build/struct.CxxQtBuilder.html#method.qml_module

**Reply by ahayzen-kdab** [reply]
Sorry yes that should read `qrc:/qt/qml/com/kdab/ui/MyButton.qml`
